setup.py

```diff
@@ -23,6 +23,18 @@ def msvc_extra_compile_args(compile_args):
     return list(chain(compile_args, xs))
 
 
+msvc_define_macros_config = [
+    ("_CRT_NONSTDC_NO_WARNINGS", None),
+    ("_CRT_SECURE_NO_WARNINGS", None),
+]
+
+
+def msvc_define_macros(macros):
+    mns = set([i[0] for i in macros])
+    xs = filter(lambda x: x[0] not in mns, msvc_define_macros_config)
+    return list(chain(macros, xs))
```

Comment:
MSVC will warn for POSIX names.
It will also warn for functions that have alternatives in Annex K.

https://learn.microsoft.com/cpp/error-messages/compiler-warnings/compiler-warning-level-3-c4996?view=msvc-170#posix-function-names
https://learn.microsoft.com/cpp/error-messages/compiler-warnings/compiler-warning-level-3-c4996?view=msvc-170#unsafe-crt-library-functions

Suppresses the warning because it is impractical to write POSIX-compatible code without using these functions.